src/node_util.cc

```diff
@@ -37,6 +39,21 @@ using v8::Value;
   VALUE_METHOD_MAP(V)
 #undef V
 
+static void GetProxyDetails(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  // Return nothing if it's not a proxy.
+  if (!args[0]->IsProxy())
+    return;
+
+  Local<Proxy> proxy = args[0].As<Proxy>();
+
+  Local<Object> ret = Object::New(env->isolate());
+  ret->Set(env->handler_string(), proxy->GetHandler());
+  ret->Set(env->target_string(), proxy->GetTarget());
```

Comment:
`Boolean::New(env->isolate(), proxy->IsRevoked())` might be of interest as well, although I guess you can also infer that by checking if the handler is non-null.

Aside, it might be a little cheaper to return an array instead of a dictionary object.